2024.2 breaks Mullvad service on Ubuntu 22.04

### Is it a bug?

- [X] I know this is an issue with the app, and contacting Mullvad support is not relevant.

### I have checked if others have reported this already

- [X] I have checked the issue tracker to see if others have reported similar issues.

### Current Behavior

Installing version 2024.2 either from the Mullvad deb repo or the deb direct download of the website results in a broken Mullvad service and thus renders the VPN unusable. The Desktop app shows a red padlock and when the UI is opened it informs the user that there is an issue with the Mullvad service causing the app to malfunction. If lockdown mode was active no connection to the internet can be made and there is no way to disable it from within the Desktop app in this state. A restart of the system does not fix the issue, neither does reinstalling the 2024.2 update. Mullvad must be uninstalled to enable internet access. Installing 2024.01 results in a working Mullvad installation.

This behavior has been confirmed on two separate machines.

### Expected Behavior

Version 2024.2 should result in a working condition Mullvad client.

### Steps to Reproduce

1. Install Mullvad 2024.02 either via Ubuntu deb repository or via the deb file downloaded from mullvad.net.
2. Mullvad service breaks, internet connection gets lost and Mullvad Dekstop client informs about issues with the Mullvad service.


### Failure Logs

```shell
System information:
id: 2c26699e-ea4c-456c-86cf-f1fa7d0a3378
kernel: 6.5.0-28-generic
mullvad-product-version: 2024.2
nm: 1.36.6
os: Linux Ubuntu 22.04.4 LTS (Jammy Jellyfish)
systemd: systemd 249 (249.11-0ubuntu3.12)
wireguard: 1.0.0

====================
Log: /var/log/mullvad-vpn/early-boot-fw.log
====================
[2024-04-29 19:41:21.135][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.2 2024-04-25
[2024-04-29 19:41:21.135][mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
[2024-04-29 19:41:21.135][mullvad_daemon::early_boot_firewall][INFO] Applying firewall policy Blocked. Blocking LAN. Allowing endpoint: none
[2024-04-29 19:41:21.135][talpid_core::firewall][INFO] Applying firewall policy: Blocked. Blocking LAN. Allowing endpoint: none

====================
Log: /var/log/mullvad-vpn/old-install-daemon.log
====================
[2024-04-29 19:39:33.013][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.1 2024-03-21
[2024-04-29 19:39:33.013][mullvad_daemon][INFO] Logging to /var/log/mullvad-vpn
[2024-04-29 19:39:33.018][mullvad_daemon::rpc_uniqueness_check][DEBUG] Failed to locate/connect to another daemon instance, assuming there isn't one
[2024-04-29 19:39:33.027][mullvad_daemon][INFO] Management interface listening on /var/run/mullvad-vpn
[2024-04-29 19:39:33.029][mullvad_api::address_cache][DEBUG] Loading API addresses from /var/cache/mullvad-vpn/api-ip-address.txt
[2024-04-29 19:39:33.041][mullvad_api::address_cache][DEBUG] Using API address: [REDACTED]:443
[2024-04-29 19:39:33.049][mullvad_api::availability][DEBUG] Suspending API requests
[2024-04-29 19:39:33.077][mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
[2024-04-29 19:39:33.083][mullvad_daemon::version_check][DEBUG] Loading version check cache from /var/cache/mullvad-vpn/version-info.json
[2024-04-29 19:39:33.086][mullvad_relay_selector][DEBUG] Reading relays from /var/cache/mullvad-vpn/relays.json
[2024-04-29 19:39:33.110][mullvad_relay_selector][DEBUG] Reading relays from /opt/Mullvad VPN/resources/relays.json
[2024-04-29 19:39:33.134][mullvad_relay_selector][INFO] Initialized with 666 cached relays from 2024-04-29 19:10:59.261
[2024-04-29 19:39:33.138][mullvad_api::availability][DEBUG] Pausing background API requests
[2024-04-29 19:39:33.141][mullvad_daemon::account_history][INFO] Opening account history file in /etc/mullvad-vpn/account-history.json
[2024-04-29 19:39:33.141][mullvad_daemon::target_state][DEBUG] No cached target state to load
[2024-04-29 19:39:33.150][talpid_core::firewall][INFO] Applying firewall policy: Blocked. Blocking LAN. Allowing endpoint: [REDACTED]:443/TCP
[2024-04-29 19:39:33.184][mullvad_api::availability][DEBUG] Pausing API requests due to being offline
[2024-04-29 19:39:33.188][mullvad_api::availability][DEBUG] Unsuspending API requests
[2024-04-29 19:39:33.188][mullvad_api::access][DEBUG] Fetching access token for an account
[2024-04-29 19:39:33.188][mullvad_daemon::geoip][DEBUG] Fetching GeoIpLocation
[2024-04-29 19:39:37.852][mullvad_api::availability][DEBUG] Resuming API requests due to coming online
[2024-04-29 19:39:37.852][mullvad_api::access][DEBUG] Fetching access token for an account
[2024-04-29 19:39:37.990][mullvad_api::availability][DEBUG] Resuming background API requests
[2024-04-29 19:39:38.010][mullvad_relay_selector::updater][DEBUG] Relay list is up-to-date
[2024-04-29 19:39:43.920][mullvad_daemon::management_interface][DEBUG] is_performing_post_upgrade
[2024-04-29 19:39:43.932][mullvad_daemon::management_interface][DEBUG] get_account_history
[2024-04-29 19:39:43.937][mullvad_daemon::management_interface][DEBUG] get_tunnel_state
[2024-04-29 19:39:44.276][mullvad_daemon::management_interface][DEBUG] get_device
[2024-04-29 19:39:44.286][mullvad_daemon::management_interface][DEBUG] update_device
[2024-04-29 19:39:44.286][mullvad_daemon::management_interface][DEBUG] get_account_history
[2024-04-29 19:39:44.287][mullvad_daemon::management_interface][DEBUG] get_settings
[2024-04-29 19:39:44.288][mullvad_daemon::management_interface][DEBUG] get_account_data
[2024-04-29 19:39:44.300][mullvad_daemon::management_interface][DEBUG] get_current_api_access_method
[2024-04-29 19:39:44.305][mullvad_daemon::management_interface][DEBUG] get_relay_locations
[2024-04-29 19:39:44.328][mullvad_daemon::device][DEBUG] The current device is still valid
[2024-04-29 19:39:44.348][mullvad_daemon::management_interface][DEBUG] get_current_version
[2024-04-29 19:39:44.350][mullvad_daemon::management_interface][DEBUG] connect_tunnel
[2024-04-29 19:39:44.350][mullvad_daemon::management_interface][DEBUG] get_version_info
[2024-04-29 19:39:44.350][mullvad_daemon][DEBUG] Target state Unsecured => Secured
[2024-04-29 19:39:44.351][mullvad_relay_selector][INFO] Selected relay ch-zrh-wg-507 at [REDACTED]
[2024-04-29 19:39:44.351][mullvad_daemon::tunnel][DEBUG] Same IP is NOT being used
[2024-04-29 19:39:44.352][talpid_core::firewall][INFO] Applying firewall policy: Connecting to [REDACTED]:20523/UDP, Blocking LAN, interface: none. Allowing endpoint [REDACTED]:443/TCP
[2024-04-29 19:39:44.383][mullvad_daemon][DEBUG] New tunnel state: Connecting { endpoint: TunnelEndpoint { endpoint: Endpoint { address: [REDACTED]:20523, protocol: Udp }, tunnel_type: Wireguard, quantum_resistant: true, proxy: None, obfuscation: None, entry_endpoint: None, tunnel_interface: None }, location: Some(GeoIpLocation { ipv4: None, ipv6: None, country: "Switzerland", city: Some("Zurich"), latitude: 47.366667, longitude: 8.55, mullvad_exit_ip: true, hostname: Some("ch-zrh-wg-507"), bridge_hostname: None, entry_hostname: None, obfuscator_hostname: None }) }
[2024-04-29 19:39:44.384][mullvad_daemon][DEBUG] Settings: openvpn mssfix: unset, wg mtu: 1420, wg ip version: IPv4, multihop: off, ipv6 (tun): off, lan: off, pq: on, obfs: off, dns: ads trackers malware
[2024-04-29 19:39:44.384][talpid_wireguard][DEBUG] Tunnel MTU: 1420
[2024-04-29 19:39:44.440][talpid_wireguard][DEBUG] Using kernel WireGuard implementation
[2024-04-29 19:39:44.442][talpid_core::firewall][INFO] Applying firewall policy: Connecting to [REDACTED]:20523/UDP over "wg0-mullvad" (ip: [REDACTED], v4 gw: [REDACTED], v6 gw: None, allowed in-tunnel traffic: [REDACTED]:1337/TCP), Blocking LAN. Allowing endpoint [REDACTED]:443/TCP
[2024-04-29 19:39:44.467][talpid_routing::imp::imp][DEBUG] Adding routes: {RequiredRoute { prefix: V4(Ipv4Network { addr: [REDACTED], prefix: 32 }), node: RealNode(Node { ip: None, device: Some("wg0-mullvad") }), main_table: true, mtu: None }}
[2024-04-29 19:39:44.468][talpid_core::firewall][INFO] Applying firewall policy: Connecting to [REDACTED]:20523/UDP over "wg0-mullvad" (ip: [REDACTED], v4 gw: [REDACTED], v6 gw: None, allowed in-tunnel traffic: [REDACTED]:1337/TCP), Blocking LAN. Allowing endpoint [REDACTED]:443/TCP
[2024-04-29 19:39:44.511][talpid_wireguard][DEBUG] Performing PQ-safe PSK exchange
[2024-04-29 19:39:44.760][talpid_tunnel_config_client][DEBUG] Config client socket MSS: 496
[2024-04-29 19:39:44.883][talpid_wireguard][DEBUG] Successfully exchanged PSK with exit peer
[2024-04-29 19:39:44.903][talpid_core::firewall][INFO] Applying firewall policy: Connecting to [REDACTED]:20523/UDP over "wg0-mullvad" (ip: [REDACTED], v4 gw: [REDACTED], v6 gw: None, allowed in-tunnel traffic: All), Blocking LAN. Allowing endpoint [REDACTED]:443/TCP
[2024-04-29 19:39:44.982][talpid_routing::imp::imp][DEBUG] Adding routes: {RequiredRoute { prefix: V4(Ipv4Network { addr: [REDACTED], prefix: 0 }), node: RealNode(Node { ip: None, device: Some("wg0-mullvad") }), main_table: false, mtu: None }}
[2024-04-29 19:39:44.982][talpid_core::firewall][INFO] Applying firewall policy: Connected to [REDACTED]:20523/UDP over "wg0-mullvad" (ip: [REDACTED], v4 gw: [REDACTED], v6 gw: None), Blocking LAN
[2024-04-29 19:39:45.011][talpid_core::dns][INFO] Setting DNS servers to [REDACTED]
[2024-04-29 19:39:45.012][talpid_core::dns::imp][DEBUG] Managing DNS via systemd-resolved
[2024-04-29 19:39:45.015][mullvad_daemon][DEBUG] New tunnel state: Connected { endpoint: TunnelEndpoint { endpoint: Endpoint { address: [REDACTED]:20523, protocol: Udp }, tunnel_type: Wireguard, quantum_resistant: true, proxy: None, obfuscation: None, entry_endpoint: None, tunnel_interface: Some("wg0-mullvad") }, location: Some(GeoIpLocation { ipv4: None, ipv6: None, country: "Switzerland", city: Some("Zurich"), latitude: 47.366667, longitude: 8.55, mullvad_exit_ip: true, hostname: Some("ch-zrh-wg-507"), bridge_hostname: None, entry_hostname: None, obfuscator_hostname: None }) }
[2024-04-29 19:39:45.015][mullvad_daemon::geoip][DEBUG] Fetching GeoIpLocation
[2024-04-29 19:39:45.215][mullvad_daemon][DEBUG] Location from am.i.mullvad.net belongs to an outdated tunnel state
[2024-04-29 19:41:06.148][mullvad_daemon::management_interface][DEBUG] prepare_restart
[2024-04-29 19:41:06.153][mullvad_daemon::shutdown::platform][DEBUG] Process received signal: [Term]
[2024-04-29 19:41:06.154][talpid_core::dns][INFO] Resetting DNS
[2024-04-29 19:41:06.156][talpid_routing::imp::imp][DEBUG] Clearing routes
[2024-04-29 19:41:06.156][mullvad_daemon][DEBUG] New tunnel state: Disconnecting(Nothing)
[2024-04-29 19:41:06.363][talpid_core::tunnel_state_machine::connecting_state][DEBUG] Tunnel monitor exited with block reason: None
[2024-04-29 19:41:06.363][talpid_core::firewall][INFO] Applying firewall policy: Blocked. Blocking LAN. Allowing endpoint: [REDACTED]:443/TCP
[2024-04-29 19:41:06.395][mullvad_daemon][DEBUG] New tunnel state: Disconnected { location: None, locked_down: true }
[2024-04-29 19:41:06.395][mullvad_daemon::geoip][DEBUG] Fetching GeoIpLocation
[2024-04-29 19:41:06.395][mullvad_daemon::device][DEBUG] Account manager has stopped
[2024-04-29 19:41:06.395][talpid_core::dns][INFO] Resetting DNS
[2024-04-29 19:41:06.395][talpid_core::tunnel_state_machine][DEBUG] Exiting tunnel state machine loop
[2024-04-29 19:41:06.396][talpid_core::tunnel_state_machine][INFO] Tunnel state machine shut down
[2024-04-29 19:41:06.396][mullvad_daemon][INFO] Mullvad daemon is quitting
[2024-04-29 19:41:06.396][mullvad_daemon::management_interface][INFO] Management interface shut down
[2024-04-29 19:41:06.896][mullvad_daemon][DEBUG] Process exiting with code 0

====================
Log: /var/log/mullvad-vpn/daemon.log
====================
[2024-04-29 19:41:21.143][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.2 2024-04-25
[2024-04-29 19:41:21.143][mullvad_daemon][INFO] Logging to /var/log/mullvad-vpn
[2024-04-29 19:41:21.143][mullvad_daemon][INFO] Management interface listening on /var/run/mullvad-vpn
[2024-04-29 19:41:21.143][mullvad_api::address_cache][DEBUG] Loading API addresses from /var/cache/mullvad-vpn/api-ip-address.txt
[2024-04-29 19:41:21.143][mullvad_api::address_cache][DEBUG] Using API address: [REDACTED]:443
[2024-04-29 19:41:21.143][mullvad_api::availability][DEBUG] Suspending API requests
[2024-04-29 19:41:21.144][mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
[2024-04-29 19:41:21.144][mullvad_daemon::version_check][DEBUG] Loading version check cache from /var/cache/mullvad-vpn/version-info.json
[2024-04-29 19:41:21.144][mullvad_daemon::version_check][WARN] Error: Unable to load cached version info
Caused by: Clearing version check cache due to a version mismatch
[2024-04-29 19:41:21.144][mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /var/cache/mullvad-vpn/relays.json
[2024-04-29 19:41:21.144][mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /opt/Mullvad VPN/resources/relays.json
[2024-04-29 19:41:21.148][mullvad_relay_selector::relay_selector][INFO] Initialized with 666 cached relays from 2024-04-25 16:03:35.000
[2024-04-29 19:41:21.148][mullvad_api::availability][DEBUG] Pausing background API requests
[2024-04-29 19:41:21.148][mullvad_daemon::account_history][INFO] Opening account history file in /etc/mullvad-vpn/account-history.json
[2024-04-29 19:41:21.148][mullvad_daemon::target_state][DEBUG] No cached target state to load
[2024-04-29 19:41:21.149][mullvad_daemon][ERROR] Error: Unable to initialize daemon
Caused by: Tunnel state machine error
Caused by: Unable to spawn offline state monitor
Caused by: The route manager returned an error
Caused by: Internal route manager error
Caused by: Failed to get a route for the given IP address
Caused by: Received a netlink error message Operation not supported (os error 95)
[2024-04-29 19:41:21.149][mullvad_daemon][DEBUG] Process exiting with code 1
[2024-04-29 19:41:21.149][mullvad_daemon::device][DEBUG] Account manager has stopped
[2024-04-29 19:41:21.149][mullvad_daemon::management_interface][INFO] Management interface shut down

====================
Log: /var/log/mullvad-vpn/early-boot-fw.old.log
====================
[2024-04-29 19:41:19.896][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.2 2024-04-25
[2024-04-29 19:41:19.896][mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
[2024-04-29 19:41:19.896][mullvad_daemon::early_boot_firewall][INFO] Applying firewall policy Blocked. Blocking LAN. Allowing endpoint: none
[2024-04-29 19:41:19.896][talpid_core::firewall][INFO] Applying firewall policy: Blocked. Blocking LAN. Allowing endpoint: none

====================
Log: /var/log/mullvad-vpn/daemon.old.log
====================
[2024-04-29 19:41:19.904][mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.2 2024-04-25
[2024-04-29 19:41:19.904][mullvad_daemon][INFO] Logging to /var/log/mullvad-vpn
[2024-04-29 19:41:19.904][mullvad_daemon][INFO] Management interface listening on /var/run/mullvad-vpn
[2024-04-29 19:41:19.904][mullvad_api::address_cache][DEBUG] Loading API addresses from /var/cache/mullvad-vpn/api-ip-address.txt
[2024-04-29 19:41:19.904][mullvad_api::address_cache][DEBUG] Using API address: [REDACTED]:443
[2024-04-29 19:41:19.904][mullvad_api::availability][DEBUG] Suspending API requests
[2024-04-29 19:41:19.905][mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
[2024-04-29 19:41:19.905][mullvad_daemon::version_check][DEBUG] Loading version check cache from /var/cache/mullvad-vpn/version-info.json
[2024-04-29 19:41:19.905][mullvad_daemon::version_check][WARN] Error: Unable to load cached version info
Caused by: Clearing version check cache due to a version mismatch
[2024-04-29 19:41:19.906][mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /var/cache/mullvad-vpn/relays.json
[2024-04-29 19:41:19.906][mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /opt/Mullvad VPN/resources/relays.json
[2024-04-29 19:41:19.909][mullvad_relay_selector::relay_selector][INFO] Initialized with 666 cached relays from 2024-04-25 16:03:35.000
[2024-04-29 19:41:19.909][mullvad_api::availability][DEBUG] Pausing background API requests
[2024-04-29 19:41:19.909][mullvad_daemon::account_history][INFO] Opening account history file in /etc/mullvad-vpn/account-history.json
[2024-04-29 19:41:19.909][mullvad_daemon::target_state][DEBUG] No cached target state to load
[2024-04-29 19:41:19.910][mullvad_daemon][ERROR] Error: Unable to initialize daemon
Caused by: Tunnel state machine error
Caused by: Unable to spawn offline state monitor
Caused by: The route manager returned an error
Caused by: Internal route manager error
Caused by: Failed to get a route for the given IP address
Caused by: Received a netlink error message Operation not supported (os error 95)
[2024-04-29 19:41:19.910][mullvad_daemon][DEBUG] Process exiting with code 1

====================
Log: ~/.config/Mullvad VPN/logs/frontend-main.log
====================
[2024-04-29 19:41:19.886][verbose] Chromium sandbox is enabled
[2024-04-29 19:41:19.887][info] Running version 2024.2
[2024-04-29 19:41:19.997][info] Detected locale: en
[2024-04-29 19:41:29.998][error] Failed to connect to daemon: Failed to connect before the deadline
[2024-04-29 19:41:29.998][error] Unable to connect to daemon: Failed to connect before the deadline

====================
Log: ~/.config/Mullvad VPN/logs/frontend-renderer.old.log
====================
[2024-04-29 19:39:45.087][verbose] Tunnel state: connected
[2024-04-29 19:39:45.088][verbose] Initial: Wrong content height: 593, expected 568
[2024-04-29 19:39:45.234][verbose] Map canvas scale factor: 1, using: 1
[2024-04-29 19:39:45.267][verbose] Tunnel state: connected
[2024-04-29 19:39:45.583][verbose] Tunnel state: connected

====================
Log: ~/.config/Mullvad VPN/logs/frontend-main.old.log
====================
[2024-04-29 19:39:43.189][verbose] Chromium sandbox is enabled
[2024-04-29 19:39:43.191][info] Running version 2024.1
[2024-04-29 19:39:43.539][info] Detected locale: en
[2024-04-29 19:39:43.904][info] Connected to the daemon
[2024-04-29 19:39:44.349][info] Autoconnect the tunnel
[2024-04-29 19:41:05.575][info] before-quit received
[2024-04-29 19:41:05.576][info] Quit initiated
[2024-04-29 19:41:05.576][info] Unsubscribed from the daemon events
[2024-04-29 19:41:05.577][verbose] GRPC Channel connectivity state changed to 4

====================
Log: ~/.config/Mullvad VPN/logs/frontend-renderer.log
====================
[2024-04-29 19:41:20.533][verbose] Tunnel state: disconnected
[2024-04-29 19:41:20.534][verbose] Initial: Wrong content height: 593, expected 568
```


### Operating system version

Ubuntu 22.04

### Mullvad VPN app version

2024.2

### Additional Information

_No response_

Hello,
Same error on debian 12.

```
journalctl -u mullvad-daemon.service

avril 30 03:37:09 piser systemd[1]: Started mullvad-daemon.service - Mullvad VPN daemon.
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon::version][INFO] Starting mullvad-daemon - 2024.2 2024-04-25
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon][INFO] Logging to /var/log/mullvad-vpn
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon][INFO] Management interface listening on /var/run/mullvad-vpn
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_api::address_cache][DEBUG] Loading API addresses from /var/cache/mullvad-vpn/api-ip-address.txt
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_api::address_cache][DEBUG] Using API address: 45.83.223.196:443
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_api::availability][DEBUG] Suspending API requests
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon::settings][INFO] Loading settings from /etc/mullvad-vpn/settings.json
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon::version_check][DEBUG] Loading version check cache from /var/cache/mullvad-vpn/version-info.json
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon::version_check][WARN] Error: Unable to load cached version info
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Failed to open app version cache file for reading
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: No such file or directory (os error 2)
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /var/cache/mullvad-vpn/relays.json
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_relay_selector::relay_selector::parsed_relays][DEBUG] Reading relays from /opt/Mullvad VPN/resources/relays.json
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_relay_selector::relay_selector][INFO] Initialized with 666 cached relays from 2024-04-25 14:33:17.000
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_api::availability][DEBUG] Pausing background API requests
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon::account_history][INFO] Opening account history file in /etc/mullvad-vpn/account-history.json
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon][INFO] Automatically connecting since auto-connect is turned on
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon][ERROR] Error: Unable to initialize daemon
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Tunnel state machine error
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Unable to spawn offline state monitor
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: The route manager returned an error
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Internal route manager error
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Failed to get a route for the given IP address
avril 30 03:37:09 piser mullvad-daemon[1522]: Caused by: Received a netlink error message Operation not supported (os error 95)
avril 30 03:37:09 piser mullvad-daemon[1522]: [mullvad_daemon][DEBUG] Process exiting with code 1
avril 30 03:37:09 piser mullvad-daemon[1522]: [43B blob data]
avril 30 03:37:09 piser systemd[1]: mullvad-daemon.service: Main process exited, code=exited, status=1/FAILURE
avril 30 03:37:09 piser systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
avril 30 03:37:10 piser systemd[1]: mullvad-daemon.service: Scheduled restart job, restart counter is at 5.
avril 30 03:37:10 piser systemd[1]: Stopped mullvad-daemon.service - Mullvad VPN daemon.
avril 30 03:37:10 piser systemd[1]: mullvad-daemon.service: Start request repeated too quickly.
avril 30 03:37:10 piser systemd[1]: mullvad-daemon.service: Failed with result 'exit-code'.
avril 30 03:37:10 piser systemd[1]: Failed to start mullvad-daemon.service - Mullvad VPN daemon.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -28,6 +28,9 @@ Line wrap the file at 100 chars.                                              Th
 #### macOS
 - Add support for split tunneling (beta).
 
+### Changed
+- Don't fail to start daemon when the offline monitor fails to be initialized.
+
 ### Fixed
 #### Linux
 - Fix GUI not working on Ubuntu 24.04 by adding an AppArmor profile.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -34,6 +34,7 @@ Line wrap the file at 100 chars.                                              Th
 ### Fixed
 #### Linux
 - Fix GUI not working on Ubuntu 24.04 by adding an AppArmor profile.
+- Service failed to start when IPv6 was disabled in the kernel.
 
 
 ## [2024.2] - 2024-04-29

--- a/talpid-core/src/offline/linux.rs
+++ b/talpid-core/src/offline/linux.rs
@@ -28,15 +28,7 @@ const PUBLIC_INTERNET_ADDRESS_V6: IpAddr =
 
 impl MonitorHandle {
     pub async fn connectivity(&self) -> Connectivity {
-        public_ip_unreachable(&self.route_manager, self.fwmark)
-            .await
-            .unwrap_or_else(|err| {
-                log::error!(
-                    "Failed to verify offline state: {}. Presuming connectivity",
-                    err
-                );
-                Connectivity::PresumeOnline
-            })
+        check_connectivity(&self.route_manager, self.fwmark).await
     }
 }
 
@@ -45,7 +37,7 @@ pub async fn spawn_monitor(
     route_manager: RouteManagerHandle,
     fwmark: Option<u32>,
 ) -> Result<MonitorHandle> {
-    let mut is_offline = public_ip_unreachable(&route_manager, fwmark).await?;
+    let mut connectivity = check_connectivity(&route_manager, fwmark).await;
 
     let mut listener = route_manager
         .change_listener()
@@ -64,18 +56,10 @@ pub async fn spawn_monitor(
         while let Some(_event) = listener.next().await {
             match sender.upgrade() {
                 Some(sender) => {
-                    let new_offline_state = public_ip_unreachable(&route_manager, fwmark)
-                        .await
-                        .unwrap_or_else(|err| {
-                            log::error!(
-                                "{}",
-                                err.display_chain_with_msg("Failed to infer offline state")
-                            );
-                            Connectivity::PresumeOnline
-                        });
-                    if new_offline_state != is_offline {
-                        is_offline = new_offline_state;
-                        let _ = sender.unbounded_send(is_offline);
+                    let new_connectivity = check_connectivity(&route_manager, fwmark).await;
+                    if new_connectivity != connectivity {
+                        connectivity = new_connectivity;
+                        let _ = sender.unbounded_send(connectivity);
                     }
                 }
                 None => return,
@@ -86,20 +70,36 @@ pub async fn spawn_monitor(
     Ok(monitor_handle)
 }
 
-async fn public_ip_unreachable(
-    handle: &RouteManagerHandle,
-    fwmark: Option<u32>,
-) -> Result<Connectivity> {
+async fn check_connectivity(handle: &RouteManagerHandle, fwmark: Option<u32>) -> Connectivity {
     let route_exists = |destination| async move {
         handle
             .get_destination_route(destination, fwmark)
             .await
-            .map_err(Error::RouteManagerError)
             .map(|route| route.is_some())
     };
-    let connectivity = Connectivity::Status {
-        ipv4: route_exists(PUBLIC_INTERNET_ADDRESS_V4).await?,
-        ipv6: route_exists(PUBLIC_INTERNET_ADDRESS_V6).await?,
-    };
-    Ok(connectivity)
+
+    match (
+        route_exists(PUBLIC_INTERNET_ADDRESS_V4).await,
+        route_exists(PUBLIC_INTERNET_ADDRESS_V6).await,
+    ) {
+        (Ok(ipv4), Ok(ipv6)) => Connectivity::Status { ipv4, ipv6 },
+        // If we fail to retrieve the IPv4 route, always assume we're connected
+        (Err(err), _) => {
+            log::error!(
+                "Failed to verify offline state: {}. Presuming connectivity",
+                err
+            );
+            Connectivity::PresumeOnline
+        }
+        // Errors for IPv6 likely mean it's disabled, so assume it's unavailable
+        (Ok(ipv4), Err(err)) => {
+            log::trace!(
+                "{}",
+                err.display_chain_with_msg(
+                    "Failed to infer offline state for IPv6. Assuming it's unavailable"
+                )
+            );
+            Connectivity::Status { ipv4, ipv6: false }
+        }
+    }
 }

--- a/talpid-core/src/offline/mod.rs
+++ b/talpid-core/src/offline/mod.rs
@@ -5,6 +5,7 @@ use talpid_routing::RouteManagerHandle;
 #[cfg(target_os = "android")]
 use talpid_types::android::AndroidContext;
 use talpid_types::net::Connectivity;
+use talpid_types::ErrorExt;
 
 #[cfg(target_os = "macos")]
 #[path = "macos.rs"]
@@ -29,8 +30,6 @@ static FORCE_DISABLE_OFFLINE_MONITOR: Lazy<bool> = Lazy::new(|| {
         .unwrap_or(false)
 });
 
-pub use self::imp::Error;
-
 pub struct MonitorHandle(Option<imp::MonitorHandle>);
 
 impl MonitorHandle {
@@ -42,39 +41,33 @@ impl MonitorHandle {
     }
 }
 
-#[cfg(not(target_os = "android"))]
 pub async fn spawn_monitor(
     sender: UnboundedSender<Connectivity>,
-    route_manager: RouteManagerHandle,
+    #[cfg(not(target_os = "android"))] route_manager: RouteManagerHandle,
     #[cfg(target_os = "linux")] fwmark: Option<u32>,
-) -> Result<MonitorHandle, Error> {
+    #[cfg(target_os = "android")] android_context: AndroidContext,
+) -> MonitorHandle {
     let monitor = if *FORCE_DISABLE_OFFLINE_MONITOR {
         None
     } else {
-        Some(
-            imp::spawn_monitor(
-                sender,
-                route_manager,
-                #[cfg(target_os = "linux")]
-                fwmark,
-            )
-            .await?,
+        imp::spawn_monitor(
+            sender,
+            #[cfg(not(target_os = "android"))]
+            route_manager,
+            #[cfg(target_os = "linux")]
+            fwmark,
+            #[cfg(target_os = "android")]
+            android_context,
         )
+        .await
+        .inspect_err(|error| {
+            log::warn!(
+                "{}",
+                error.display_chain_with_msg("Failed to spawn offline monitor")
+            );
+        })
+        .ok()
     };
 
-    Ok(MonitorHandle(monitor))
-}
-
-#[cfg(target_os = "android")]
-pub async fn spawn_monitor(
-    sender: UnboundedSender<Connectivity>,
-    android_context: AndroidContext,
-) -> Result<MonitorHandle, Error> {
-    let monitor = if *FORCE_DISABLE_OFFLINE_MONITOR {
-        None
-    } else {
-        Some(imp::spawn_monitor(sender, android_context).await?)
-    };
-
-    Ok(MonitorHandle(monitor))
+    MonitorHandle(monitor)
 }

--- a/talpid-core/src/tunnel_state_machine/mod.rs
+++ b/talpid-core/src/tunnel_state_machine/mod.rs
@@ -55,10 +55,6 @@ const TUNNEL_STATE_MACHINE_SHUTDOWN_TIMEOUT: Duration = Duration::from_secs(5);
 /// Errors that can happen when setting up or using the state machine.
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    /// Unable to spawn offline state monitor
-    #[error("Unable to spawn offline state monitor")]
-    OfflineMonitorError(#[from] crate::offline::Error),
-
     /// Unable to set up split tunneling
     #[cfg(any(target_os = "windows", target_os = "macos"))]
     #[error("Failed to initialize split tunneling")]
@@ -339,8 +335,7 @@ impl TunnelStateMachine {
             #[cfg(target_os = "android")]
             android_context,
         )
-        .await
-        .map_err(Error::OfflineMonitorError)?;
+        .await;
         let connectivity = offline_monitor.connectivity().await;
         let _ = initial_offline_state_tx.unbounded_send(connectivity);
 

